### PR TITLE
Add option for verbose biome data

### PR DESCRIPTION
**Pull request by Montandalar**
and rebase off of the latest master.

If you would rather I keep this branch maintained in my own fork instead, that is fine, just close the PR.

### CODE REVIEW

**Comment by Montandalar** [comment]
Hmm I hadn't considered the effect of a merge conflict coming from the rebase either. What do you call that? A rebase request instead of a pull request?

File: huds.lua [hunk 1]
```diff
@@ -42,8 +43,17 @@ function postool.readPlayerToggles(oPlayer)
 	end
 
 	-- add chunk border toggle for tool usage, if not yet existing
-	if 6 == #sFlags then sFlags[7] = false end
+	if 6 == #tb then tb[7] = false end
```
Comment (SwissalpS) [inline huds.lua:46]: this line makes no sense, it should be deleted.

one conflict less after that

File: functions.lua [hunk 2]
```diff
@@ -66,6 +66,50 @@ postool.hasMeseconsDebug = function()
 end -- hasMeseconsDebug
 
 
+-- return a string to show on HUD containing info about biome that player is standing on
+-- thanks to Montandalar for initial method
+if postool.hudBiomeVerbose then
```
Comment (SwissalpS) [inline functions.lua:71]: this check should be done inside the function then wouldn't need to have two functions to maintain.

**Comment by SwissalpS** [comment]
also, don't you want user to be able to decide between verbose info and compact?

File: settingtypes.txt [hunk 3]
```diff
@@ -30,11 +32,14 @@ postool.hud.defaultshowblock (Show block coordinates by default) bool true
...
 # wait at least this long before updating HUD (seconds)
 postool.hud.minupdateinterval (Update HUD interval) int 2 1 999
 # wait at least this long before grid vanishes
 postool.tool.griddisplayduration (Grid display duration) int 12 1 999
 # don't show chunk border indicator. Only shows up when not in center block.
 # Setting this to true makes it never show at all.
 postool.tool.suppresschunkindicator (Hide chunk border indicator) bool false
-
+# Show the heat and humidity values of your position to two decimal places.
+postool.biomedata.verbose (More biome data) bool false
```
Comment (SwissalpS) [inline settingtypes.txt:45]: This is part of the HUD, the setting name should reflect that: postool.hud.biomedata.verbose or similar and as such should be placed up where it's relative settings are.

**Comment by SwissalpS** [comment]
let's clear this up before merging in changes in master

**Comment by SwissalpS** [comment]
after this commit the mentioned issues are cleared up:
https://github.com/SwissalpS/postool/commit/61f4d875a899c29d594c541278461c0808a11d98

Now we only need to add user toggle - if that is requested.

**Comment by SwissalpS** [comment]
and with commit [271f65](https://github.com/SwissalpS/postool/commit/271f65f89085dba472f6b162e137968af89d2942) we also have user toggle.
I don't like it though. Adding line feeds '\n' after main biome info and after first detail it looks ok. But only if all other toggles are turned off.

**Comment by SwissalpS** [comment]
[...and with this commit](https://github.com/SwissalpS/postool/commit/6464ff6394bdde9c04f18794eea2ad8f3775c9b0) this branch also has the changes of master branch.
Feel free to pull the changes to your branch :D

Edit: well the readme could use some updating...

**Pull request closed without merging**